### PR TITLE
Sending ready messages to Distributor is no longer fire & forget 

### PR DESCRIPTION
**Pull request by tmasternak**
@Particular/nservicebus-maintainers can you have a look?

### CODE REVIEW

File: src/NServiceBus.Core/Routing/Legacy/ReadyMessageSender.cs [hunk 1]
```diff
@@ -66,7 +67,7 @@ public void MessageProcessed(Dictionary<string, string> headers)
             }
             else
             {
-                SendReadyMessage(1, false);
+                await SendReadyMessage(1, false).ConfigureAwait(false);
```
Comment (danielmarbach) [inline src/NServiceBus.Core/Routing/Legacy/ReadyMessageSender.cs]: Here there would be an opportunity to

`return SendReadyMessage`

and in the other exit points in the method

`return TaskEx.CompletedTask`

What do you think?

Comment (tmasternak) [inline src/NServiceBus.Core/Routing/Legacy/ReadyMessageSender.cs]: :+1:

**Comment by danielmarbach** [comment]
That was a really nasty hidden one. Normally we should get a compiler warning. The problem is the following

```
public static void OuterMethod()
{
   // Will result in a compiler warning
   SomeAsyncMethod();
   // Will not result in a compiler warning
   SomeTaskReturningMethod();
}

public static async Task SomeAsyncMethod();
public static Task SomeTaskReturningMethod();
```

**Comment by danielmarbach** [comment]
How about a squash and reword to something like

Sending ready messages is no longer fire & forget (I'm sure this is a lousy text and you can come up with a better idea ;) )

and then rename the PR as well?

**Comment by danielmarbach** [comment]
@bording Can you review as well and merge?

**Comment by bording** [comment]
:+1:

**Pull request merged**